scipy/optimize/_trustregion_iterative.py

```diff
@@ -0,0 +1,470 @@
+        self.niter = 0
+        self.easycase_stop_criteria = min(trust_region_factor-1,
+                                          1-np.sqrt(optimization_factor))
+        self.hardcase_stop_criteria = 1-optimization_factor
+        self._hess_gershgorin_ub = None
+        self._hess_gershgorin_lb = None
+        self._hess_inf = None
+        self._hess_frob = None
+        self._dimension = None
+
+        # Superclass __init__
+        BaseQuadraticSubproblem.__init__(self, x, fun, jac,
+                                         hess)
+
+        # Get Lapack function for cholesky decomposition
+        self.cholesky, = get_lapack_funcs(('potrf',), (self.hess,))
+
+        # A constant such that for vectors smaler than that
+        # backward substituition is not reliable. It was stabilished
+        # based on Golub, G. H., Van Loan, C. F. (2013).
+        # "Matrix computations". Forth Edition. JHU press., p.165.
+        self.CLOSE_TO_ZERO = self.dimension*self.EPS*self.hess_inf
+
+    @property
+    def dimension(self):
+        """Hessian inf norm"""
+        if self._dimension is None:
+            self._dimension = len(self.hess)
+        return self._dimension
+
+    @property
+    def hess_gershgorin_ub(self):
+        """Gershgorin upper bound defined on p.19 from [1]
+        computed for the hessian"""
+        if self._hess_gershgorin_ub is None:
+            n = self.hess.shape[0]
+
+            # Compute H[i,i] + sum(abs(H[i, j]), i!=j)
+            maximum = -np.Inf
+            for i in range(n):
+                summation = self.hess[i, i]
+                for j in range(n):
+                    summation += abs(self.hess[i, j]) if i != j else 0
+
+                if summation > maximum:
+                    maximum = summation
+
+            self._hess_gershgorin_ub = maximum
+        return self._hess_gershgorin_ub
+
+    @property
+    def hess_gershgorin_lb(self):
+        """Gershgorin lower bound defined on p.19 from [1]
+        computed for the hessian"""
+        if self._hess_gershgorin_lb is None:
+            n = self.hess.shape[0]
+
+            # Compute -H[i,i] + sum(abs(H[i, j]), i!=j)
+            maximum = -np.Inf
+            for i in range(n):
+                summation = -self.hess[i, i]
+                for j in range(n):
+                    summation += abs(self.hess[i, j]) if i != j else 0
+
+                if summation > maximum:
+                    maximum = summation
+
+            self._hess_gershgorin_lb = -maximum
+        return self._hess_gershgorin_lb
+
+    @property
+    def hess_inf(self):
+        """Hessian inf norm"""
+        if self._hess_inf is None:
+            self._hess_inf = norm(self.hess, np.Inf)
+        return self._hess_inf
+
+    @property
+    def hess_fro(self):
+        """Hessian frobenius norm"""
+        if self._hess_frob is None:
+            self._hess_frob = norm(self.hess, 'fro')
+        return self._hess_frob
+
+    def _initial_values(self, trust_radius):
```

Comment:
Actually I have done that! The lower bound `lb` will be the maximum between the one used in the previous iteration and the computed for this iteration (in the case of sequential reductions in the trust radius).  Since the `damp` is the middle point between `lb` and `ub`it will also be affected by the previous decision.

I have done some experiments regarding this while implementing the algorithm and I got this schema to be simple and effective.